glotaran/analysis/residual_calculation.py

```diff
@@ -14,14 +19,14 @@ def create_index_independent_ungrouped_residual(
     penalties = []
     for label in problem_bag:
         data = problem_bag[label].data
-        size = problem_bag[label].global_axis.size
         weight = problem_bag[label].weight
+        global_axis = problem_bag[label].global_axis
         reduced_clp_labels[label] = constraint_labels_and_matrices[label].clp_label
         matrix = constraint_labels_and_matrices[label].matrix
 
         reduced_clps[label] = []
         residuals[label] = []
-        for i in range(size):
+        for i, index in enumerate(global_axis):
```

Comment:
Using `enumerate` and NOT using index doesn't make sense. Also, it is a code smell.

Reply:
Opps this was intended for #413